easybuild/easyconfigs/q/QUDA/QUDA-1.1.0-foss-2022a-CUDA-11.7.0.eb

```
@@ -0,0 +1,40 @@
+##
+# Author:    Robert Mijakovic <robert.mijakovic@lxp.lu>
+##
+easyblock = 'CMakeMake'
+
+name = 'QUDA'
+version = '1.1.0'
+versionsuffix = '-CUDA-%(cudaver)s'
+
+homepage = 'https://github.com/lattice/quda'
+description = """QUDA is a library for performing calculations in lattice QCD
+ on graphics processing units (GPUs), leveraging NVIDIA's CUDA platform."""
+
+toolchain = {'name': 'foss', 'version': '2022a'}
+
+github_account = 'lattice'
+source_urls = [GITHUB_SOURCE]
+sources = ['v%(version)s.tar.gz']
+checksums = ['b4f635c993275010780ea09d8e593e0713a6ca1af1db6cc86c64518714fcc745']
+
+builddependencies = [
+    ('CMake', '3.23.1'),
+    ('Python', '3.10.4'),
+]
+
+dependencies = [
+    ('CUDA', '11.7.0', '', True),
+    ('UCX-CUDA', '1.12.1', versionsuffix),
+    ('Eigen', '3.4.0'),
+    ('NVSHMEM', '2.6.0', versionsuffix),
+]
+
+configopts = '-DQUDA_GPU_ARCH=sm_80 -DQUDA_MPI=ON -DQUDA_NVSHMEM=ON -DQUDA_NVSHMEM_HOME=$EBROOTNVSHMEM -DQUDA_DOWNLOAD_EIGEN=OFF -DPROPAGATED_FLAGS=" " -DMPIEXEC_EXECUTABLE="$(which srun)" -DMPIEXEC_PREFLAGS="--overlap" -DCMAKE_EXE_LINKER_FLAGS="-L$EBROOTUCX/lib -lnvidia-ml -lucs -lucp"'
```

This line is too long, it should be broken up in multiple shorter lines, for example with:

```suggestion
configopts = '-DQUDA_GPU_ARCH=sm_80 -DQUDA_MPI=ON -DQUDA_NVSHMEM=ON -DQUDA_NVSHMEM_HOME=$EBROOTNVSHMEM '
configopts += '-DQUDA_DOWNLOAD_EIGEN=OFF -DPROPAGATED_FLAGS=" " -DMPIEXEC_EXECUTABLE="$(which srun)" '
configopts += '-DMPIEXEC_PREFLAGS="--overlap" -DCMAKE_EXE_LINKER_FLAGS="-L$EBROOTUCX/lib -lnvidia-ml -lucs -lucp"'
```

In addition:

* You should use `%(cuda_sm_comma_sep)s` (which is derived from the EasyBuild configuration setting `--cuda-` instead of hardcoding `sm_80`; see also https://docs.easybuild.io/en/latest/version-specific/easyconfig_templates.html#template-values-which-are-defined-dynamically
* What happens if `srun` is not available, i.e. when this easyconfig is installed on a system that doesn't have Slurm available? Is specifying `-DMPIEXEC_EXECUTABLE` and `-DMPIEXEC_PREFLAGS` strictly necessary?